File: src/LegacyVault.API/Models/Entities/Category.cs
using LegacyVault.API.Models.Enums;

namespace LegacyVault.API.Models.Entities;

public class Category
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public CategoryType Type { get; set; } = CategoryType.General;
    public string Name { get; set; } = string.Empty;
    public string Icon { get; set; } = "📁";
    public bool IsEncrypted { get; set; }
    public string? EncryptionSalt { get; set; }  // Base64-encoded 256-bit salt
    public string? PasswordHint { get; set; }    // Optional hint shown on unlock prompt
    public bool IsFavorite { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public ICollection<Page> Pages { get; set; } = new List<Page>();
}

File: src/LegacyVault.Android/app/src/main/kotlin/com/legacyvault/app/domain/model/Category.kt
package com.legacyvault.app.domain.model

import com.legacyvault.app.domain.model.enums.CategoryType

data class Category(
    val id: String,
    val type: CategoryType,
    val name: String,
    val icon: String,
    val isEncrypted: Boolean,
    val encryptionSalt: String?,
    val passwordHint: String?,
    val isFavorite: Boolean,
    val pageCount: Int,
    val createdAt: String,
    val updatedAt: String
) {
    val isVault: Boolean get() = type == CategoryType.Vault
}
